src/components/svg/lineDrawing.ts

```diff
@@ -195,97 +195,107 @@ export const stackedArea = ({
   yPropTop,
   yPropBottom
 }: StackedAreaTypes) => {
-  const uniqXValues = data
-    .map(d => d.data.map(p => datesForUnique(p[xProp])))
-    .reduce((a, b) => a.concat(b), [])
-    .reduce((p, c) => {
-      if (p.indexOf(c) === -1) {
-        p.push(c)
+  const a = performance.now()
+  const valueMap = new Map()
+  const lineSums = data.map(() => 0)
+
+  let lineIndex = 0
+  for (const line of data) {
+    line.__lineIndex = lineIndex
```

Comment:
How is this being used? Maybe a line can be tracked by its index in the array?

Reply:
I sort the lines after this calculation, so I need to know their original index order in order to use the naively populated coordinate array at each x. Otherwise the coordinates in the map will be in different order than the sorted order of the lines.

I'll add this as a comment.